pkg/provider/provider.go

```diff
@@ -137,3 +156,23 @@ func convertToTypeSet(r *schema.Resource) *schema.Schema {
 		Elem: r,
 	}
 }
+
+// ValidateIAMVersion is a ValidateFunc for the "iam_version" field in the provider schema
+func ValidateIAMVersion(v interface{}, k string) ([]string, []error) {
+	// check that v is in iamVersionList
+	found := false
+	for _, version := range iamVersionList {
+		if version == v.(string) {
```

Comment:
Unlikely to be an issue at the moment, but probably still worth using a two return form type assertion here(e.g. `v, ok := v.(string)`) to avoid panics.